bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -70,12 +74,18 @@ def apply_prep(target_spec)
 
           unless need_install.empty?
             need_install_targets = need_install.map(&:target)
-            run_task(executor, need_install_targets, 'puppet_agent::install')
-            # Service task works best when targets have puppet-agent feature
+
+            need_install_targets.each do |t|
+              plugin_name = t.plugin_hooks&.fetch('puppet_library')&.fetch('plugin') || 'install_agent'
+              hook = inventory.plugins.get_hook(plugin_name, :puppet_library)
+              begin
+                hook.call(t, t.plugin_hooks['puppet_library'])
```

Comment:
Am I missing something or will this will be a serious performance regression since we'll no longer parallelize the agent installation.